aea/aea.py

```diff
@@ -220,6 +220,7 @@ def __init__(
             search_service_address,
             decision_maker_handler.self_address,
             data_dir,
+            public_keys=wallet.public_keys,
```

Comment:
It might be preferable to extend the identity object instead to contain the public keys. This would be consistent with its intended usage. You'll have to extend the identity object and also the constructor here: https://github.com/fetchai/agents-aea/blob/main/aea/aea_builder.py#L1221